sgkit/cohorts.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional, Sequence, Tuple, Union
+
+import numpy as np
+import pandas as pd
+
+from sgkit.typing import ArrayLike
+
+
+def _tuple_len(t: Union[int, Tuple[int, ...], str, Tuple[str, ...]]) -> int:
+    """Return the length of a tuple, or 1 for an int or string value."""
+    if isinstance(t, int) or isinstance(t, str):
+        return 1
+    return len(t)
+
+
+def _cohorts_to_array(
+    cohorts: Sequence[Union[int, Tuple[int, ...], str, Tuple[str, ...]]],
+    index: Optional[pd.Index] = None,
+) -> ArrayLike:
+    """Convert cohorts or cohort tuples specified as a sequence of values or
+    tuples to an array of ints used to match samples in ``sample_cohorts``.
+
+    Cohorts can be specified by index (as used in ``sample_cohorts``), or a label, in
+    which case an ``index`` must be provided to find index locations for cohorts.
```

Comment:
Any chance of a couple of simple examples here and the return values? I'm finding it a bit abstract and a concrete example would help understand what the function does.

Reply:
Thanks for the review @jeromekelleher. Added two examples in https://github.com/pystatgen/sgkit/pull/394/commits/f4b45ea3cba8e234c65c43944d135dc262fa5ec0